File: code/backend-ms1/src/main/java/ma/zs/univ/service/facade/admin/NotificationRetardDeuxiemeNiv/NotificationRetardDeuxiemeNiveauDetailAdminService.java
package ma.zs.univ.service.facade.admin.NotificationRetardDeuxiemeNiv;

import java.util.List;
import ma.zs.univ.bean.core.NotificationRetardDeuxiemeNiv.NotificationRetardDeuxiemeNiveauDetail;
import ma.zs.univ.dao.criteria.core.NotificationRetardDeuxiemeNiv.NotificationRetardDeuxiemeNiveauDetailCriteria;
import ma.zs.univ.zynerator.service.IService;



public interface NotificationRetardDeuxiemeNiveauDetailAdminService extends  IService<NotificationRetardDeuxiemeNiveauDetail,NotificationRetardDeuxiemeNiveauDetailCriteria>  {

    List<NotificationRetardDeuxiemeNiveauDetail> findByTrimId(Long id);
    int deleteByTrimId(Long id);
    long countByTrimId(Long id);




}

File: code/frontend/src/app/shared/service/admin/NotificationRetardDeuxiemeNiv/NotificationRetardDeuxiemeNiveauDetailAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {NotificationRetardDeuxiemeNiveauDetailDto} from 'src/app/shared/model/NotificationRetardDeuxiemeNiv/NotificationRetardDeuxiemeNiveauDetail.model';
import {NotificationRetardDeuxiemeNiveauDetailCriteria} from 'src/app/shared/criteria/NotificationRetardDeuxiemeNiv/NotificationRetardDeuxiemeNiveauDetailCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class NotificationRetardDeuxiemeNiveauDetailAdminService extends AbstractService<NotificationRetardDeuxiemeNiveauDetailDto, NotificationRetardDeuxiemeNiveauDetailCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): NotificationRetardDeuxiemeNiveauDetailDto {
        return new NotificationRetardDeuxiemeNiveauDetailDto();
    }

    public constrcutCriteria(): NotificationRetardDeuxiemeNiveauDetailCriteria {
        return new NotificationRetardDeuxiemeNiveauDetailCriteria();
    }

    get API() {
        return environment.apiUrlUnivservice + 'admin/notificationRetardDeuxiemeNiveauDetail/';
    }
}
